src/GraphQL.Conventions/Extensions/Utilities.cs

```diff
@@ -19,5 +27,115 @@ public static class Utilities
+
+            return lambda(instance, arguments);
+        }
+        private static ConcurrentDictionary<MethodInfo, Func<object, object[], object>> _methodDictionary = new ConcurrentDictionary<MethodInfo, Func<object, object[], object>>();
+        private static Func<object, object[], object> InvokeEnhancedUncachedMethod(MethodInfo methodInfo)
+        {
+            var instanceParameter = Expression.Parameter(typeof(object));
+            var argumentsParameter = Expression.Parameter(typeof(object[]));
+            var instanceExpression = methodInfo.IsStatic ? null : Expression.Convert(instanceParameter, methodInfo.DeclaringType);
+            var methodParameters = methodInfo.GetParameters();
+            var parameters = methodParameters.Select((param, index) => {
+                return Expression.Convert(Expression.ArrayAccess(argumentsParameter, Expression.Constant(index)), param.ParameterType);
+            });
+            var call = Expression.Call(instanceExpression, methodInfo, parameters);
+            Expression body;
+            if (methodInfo.ReturnType == typeof(void))
+            {
+                body = Expression.Block(new Expression[] {
+                    call,
+                    Expression.Constant(null, typeof(object))
+                });
+            }
+            else
+            {
+                body = Expression.Convert(call, typeof(object));
+            }
+            var lambda = Expression.Lambda<Func<object, object[], object>>(body, instanceParameter, argumentsParameter);
+            return lambda.Compile();
+        }
+
+        /// <summary>
+        /// Invokes a constructor represented by a specified <see cref="ConstructorInfo"/>, using
+        /// the specified parameters.
+        /// </summary>
+        /// <param name="constructorInfo">
+        /// The constructor representation. Must not be a static constructor.
+        /// </param>
+        /// <param name="arguments">
+        /// An argument list for the invoked constructor. This is an array of objects
+        /// with the same number, order, and type as the parameters of the constructor
+        /// to be invoked. If there are no parameters, parameters should be null.
+        /// Ref/out parameters are not supported.
+        /// </param>
+        /// <returns>
+        /// The constructed object.
+        /// </returns>
+        public static object InvokeEnhanced(this ConstructorInfo constructorInfo, object[] arguments)
+        {
+            //just good practice
+            if (constructorInfo == null)
+                throw new ArgumentNullException(nameof(constructorInfo));
+            //the lambda ignores extra arguments so the following check is necessary
+            if (constructorInfo.GetParameters().Length != (arguments?.Length ?? 0))
+                throw new ArgumentException("Invalid number of arguments for this constructor", "arguments");
```

Comment:
Might be worth using a static `throw`-helper here; see: https://reubenbond.github.io/posts/dotnet-perf-tuning

>When a method contains a throw statement, the JIT will not inline it. The common trick to solve this is to add a static "throw helper" method to do the dirty work for you, so the end result looks like this: